### ISSUE TITLE
Suggestion to improve the scheduling of region Ticks

### ISSUE THREAD

**Issue by PedroMPagani** [PaperMC/Folia#257]
### Is your feature request related to a problem?

It may be, technically speaking, there's a small "gap" in knowledge, basically from what I see, there's two types of ticking, you have a global ticking per region, I am not sure if it's per region but from my debugs it was running a lot more than just once, I am not sure if this will be per level.

basically, on the current task grabbing, the thread basically parkNanos until the deadline, the problem is, a newer task with a lower deadline (meaning before your cuirrent park nanos target) may appear, which could for example, be a good resource efficiency on the usage, and you can "stop" the parkNanos, to instead be grabbing the other task, since it may run earlier. This isn't exactly a big problem, but this creates a not good behavior that is almost not easy to debug.

### Describe the solution you'd like.

I kind of gave it away on the problem above. if this is kind of modified, you could basically have 20 ticking regions,
in "single thread" ticking for example, in which otherwise it wouldn't be exactly possible.

### Describe alternatives you've considered.

The alternatives is basically one, the more regions you have, the more and more threads you'll need, even if the regions ticks are for example in a low mspt, this extra resource requirement, is FULLY expected and listed on the Folia basic hardware "requirements", the more regions, the more threads you'll need, but with this efficiency, it can be improved by like 1000x,

**Comment by PedroMPagani** [comment]
I am not experiencing this exact issue, due to the custom implementation on https://github.com/PaperMC/Folia/issues/196, the issue kind of diminishes itself, but on a high count of regions and "20 ticks" of thread per sec, meaning around 50ms gabs every task, this issue will most def appear a lot more.

**Comment by cegou121** [comment]
Problem #196 fix

**Comment by Spottedleaf** [comment]
This is already an invariant maintained by the scheduler (SchedulerThreadPool). For a thread pool of size `n`, the earliest `n` tasks are the tasks being blocked on.

**Comment by PedroMPagani** [comment]
> This is already an invariant maintained by the scheduler (SchedulerThreadPool). For a thread pool of size `n`, the earliest `n` tasks are the tasks being blocked on.

Yes, but that's earliest based on current data, if you have 128 regions, you may also want to have more than 64 threads otherwise you can run into scheduling problems?

**Comment by Spottedleaf** [comment]
It's earliest based on whichever tasks are scheduled.
